### PR TITLE
bpf: add test for encrypted overlay

### PR DESCRIPTION
**Pull request by julianwiedmann**
Add a full integration test for the new [encrypted overlay path](https://github.com/cilium/cilium/pull/31073) in `to-netdev`.

### CODE REVIEW

File: bpf/tests/tc_host_encrypted_overlay.c [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+	struct vxlanhdr *vxlan;
+	void *data, *data_end;
+	__u32 *status_code;
+	struct udphdr *l4;
+	struct ethhdr *l2;
+	struct iphdr *l3;
+
+	test_init();
+
+	data = (void *)(long)ctx_data(ctx);
```
Comment (ldelossa) [inline bpf/tests/tc_host_encrypted_overlay.c:140]: nit: 
 
My personal opinion, but we should kill the (long) casts and use the fixed size types. 
The rest of your test does. I don't think we hit any lint or compiler issues by doing this.

**Review by ldelossa — APPROVED** [review]
Test looks good! stupid nit that you can ignore.

File: bpf/tests/tc_host_encrypted_overlay.c (hunk 1 above)
Comment (julianwiedmann) [reply bpf/tests/tc_host_encrypted_overlay.c:140]: oh I bet this is just a copy&paste, and we can clean up that pattern across a whole range of tests :)

**Pull request merged**